kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStore.java

```diff
@@ -16,57 +16,36 @@
 
 package dev.responsive.kafka.store;
 
-import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.asInternalProcessorContext;
-import static org.apache.kafka.streams.processor.internals.ProcessorContextUtils.changelogFor;
-
-import com.datastax.oss.driver.api.core.cql.BoundStatement;
 import dev.responsive.db.CassandraClient;
-import dev.responsive.kafka.store.CommitBuffer.BufferPlugin;
-import dev.responsive.model.Result;
 import dev.responsive.utils.RemoteMonitor;
-import dev.responsive.utils.StoreUtil;
 import dev.responsive.utils.TableName;
-import java.time.Duration;
 import java.util.List;
-import java.util.concurrent.TimeoutException;
 import org.apache.kafka.clients.admin.Admin;
-import org.apache.kafka.clients.consumer.ConsumerRecord;
-import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability.Evolving;
+import org.apache.kafka.common.serialization.Serializer;
 import org.apache.kafka.common.utils.Bytes;
 import org.apache.kafka.streams.KeyValue;
-import org.apache.kafka.streams.errors.ProcessorStateException;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.processor.StateStoreContext;
-import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
-import org.apache.kafka.streams.processor.internals.RecordCollector;
-import org.apache.kafka.streams.processor.internals.RecordCollector.Supplier;
+import org.apache.kafka.streams.processor.internals.GlobalProcessorContextImpl;
 import org.apache.kafka.streams.query.Position;
+import org.apache.kafka.streams.query.PositionBound;
+import org.apache.kafka.streams.query.Query;
+import org.apache.kafka.streams.query.QueryConfig;
+import org.apache.kafka.streams.query.QueryResult;
 import org.apache.kafka.streams.state.KeyValueIterator;
 import org.apache.kafka.streams.state.KeyValueStore;
-import org.apache.kafka.streams.state.internals.StoreQueryUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ResponsiveStore implements KeyValueStore<Bytes, byte[]> {
 
-  private static final Logger LOG = LoggerFactory.getLogger(ResponsiveStore.class);
-
-  // visible for testing
-  static final Plugin PLUGIN = new Plugin();
-
   private final CassandraClient client;
   private final TableName name;
   private final RemoteMonitor initRemote;
   private final Admin admin;
-  private final Position position;
-
-  private boolean open;
-  private CommitBuffer<Bytes> buffer;
 
-  @SuppressWarnings("rawtypes")
-  private InternalProcessorContext context;
-  private int partition;
+  // Visible for testing
+  KeyValueStore<Bytes, byte[]> delegate;
```

Comment:
Should there be a package-private getter? Makes me uncomfortable to expose the real thing like this 😅

Reply:
🙈 I have no idea what possessed me to do it this way